client/src/components/UserForm.js

```diff
@@ -1,120 +1,181 @@
-import { useState } from "react";
-import { Link } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from "yup";
 import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
 
 import "./UserForm.css";
 
+const schema = yup.object({
+  name: yup.string().min(3).max(50).matches(/^[a-zA-Z\s]+$/, 'must be letters only.').required(),
+
+  gender: yup.string().required(),
+
+  dateOfBirth: yup.string().required("Please enter a date.").matches(/^\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])$/, 'must be a valid date in the format DD/MM/YYYY'),
```

Comment:
There is a `.date()` format in yup which could simplify a bit this line and the `dateOfRelease` [see this resource](https://www.codedaily.io/tutorials/Yup-Date-Validation-with-Custom-Transform)

Reply:
done